### PR TITLE
mr: Fix using Date type as partition field

### PR DESCRIPTION
**Pull request by lurnagao-dahua**
close [7359](https://github.com/apache/iceberg/pull/7359) and  reopen

### CODE REVIEW

**Comment by lurnagao-dahua** [comment]
Hi, Can you take a look for me? @nastra

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java [hunk 1]
```diff
@@ -1206,6 +1208,55 @@ public void testRemoveAndAddBackColumnFromIcebergTable() throws IOException {
...
+    List<Record> records =
+        TestHelper.RecordsBuilder.newInstance(dateSchema)
+            .add(1L, LocalDateTime.of(2023, 1, 21, 21, 10, 10, 100000000))
+            .add(2L, LocalDateTime.of(2023, 1, 21, 22, 10, 10, 200000000))
+            .add(3L, LocalDateTime.of(2023, 1, 22, 21, 10, 10, 300000000))
+            .build();
+    testTables.createTable(shell, "part_test", dateSchema, spec, FileFormat.PARQUET, records);
+    List<Object[]> result = shell.executeStatement("SELECT * from part_test order by id");
+
+    assertThat(records.size()).isEqualTo(result.size());
```
Comment (nastra) [inline mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:1254]:
```suggestion
    assertThat(result).hasSameSizeAs(records);
```

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java [hunk 2]
```diff
@@ -1206,6 +1208,55 @@ public void testRemoveAndAddBackColumnFromIcebergTable() throws IOException {
...
+    assertThat("2023-01-21 21:10:10.1").isEqualTo(result.get(0)[1]);
```
Comment (nastra) [inline mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:1255]: this should be `assertThat(result.get(0)[1]).isEqualTo(...)`. Same for the others

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java (hunk 1 above)
Comment (nastra) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java:1254]: please also update the same thing in the other test

File: mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java (hunk 2 above)
Comment (lurnagao-dahua) [reply mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java]: Hi, I updated.

**Review by nastra — APPROVED** [review]
LGTM, thanks for fixing this

**Comment by nastra** [comment]
@Fokko since you were already looking at https://github.com/apache/iceberg/pull/7179, could you take a look at this one as well please?

**Comment by lurnagao-dahua** [comment]
@Fokko  I am very much looking forward to and grateful for your review

**Comment by lurnagao-dahua** [comment]
@Fokko Hi, I am very much looking forward to and grateful for your review!

**Review by nastra — APPROVED** [review]

**Pull request merged**